### PR TITLE
CN-330 Fixes on share dialog.

### PR DESCRIPTION
**Pull request by mmargarit**
Added intent type and more tests for null cases

### CODE REVIEW

File: sdk/src/main/java/com/getdreams/views/DreamsView.kt [hunk 1]
```diff
@@ -354,12 +355,14 @@ class DreamsView : FrameLayout, DreamsViewInterface {
     fun openShareDialog(text: String?, title: String?, url: String?) {
         val share = Intent.createChooser(Intent().apply {
             action = Intent.ACTION_SEND
-            putExtra(Intent.EXTRA_TEXT, text)
+            putExtra(Intent.EXTRA_TEXT, Html.fromHtml(text + "<br/>" + url))
```
Comment (ninaniiranen) [inline sdk/src/main/java/com/getdreams/views/DreamsView.kt]: Why use the deprecated version of `Html.fromHtml`? And as the only tag we add is a break we can just add `\n` instead if having to parse html. We could do `if (url.isNullOrEmpty()) { text } else { "$text\n$url" }` instead.

File: sdk/src/main/java/com/getdreams/views/DreamsView.kt [hunk 2]
```diff
@@ -354,12 +355,14 @@ class DreamsView : FrameLayout, DreamsViewInterface {
     fun openShareDialog(text: String?, title: String?, url: String?) {
```
Comment (ninaniiranen) [inline sdk/src/main/java/com/getdreams/views/DreamsView.kt]: As we don't call this if text is null we can remove the nullable type on that.

**Review by ninaniiranen — CHANGES_REQUESTED** [review]

File: sdk/src/main/java/com/getdreams/views/DreamsView.kt (hunk 1 above)
Comment (mmargarit) [reply sdk/src/main/java/com/getdreams/views/DreamsView.kt]: '\n' doesn't work on share dialog, it needs html. Will check the updated version of the html.

Comment (ninaniiranen) [reply sdk/src/main/java/com/getdreams/views/DreamsView.kt]: `\n` works fine for me

**Comment by mmargarit** [comment]
Ready to re 👀

**Review by ninaniiranen — APPROVED** [review]
